srml/contracts/src/wasm/prepare.rs

```diff
@@ -74,6 +74,23 @@ impl<'a> ContractModule<'a> {
 		Ok(())
 	}
 
+	/// Ensures that tables declared in the module are not too big.
+	fn ensure_table_size_limit(&self, limit: u32) -> Result<(), &'static str> {
+		let err_msg = self.module.table_section()
+			.and_then(|table_section| table_section.entries().first())
+			.and_then(|table_type| {
+				if table_type.limits().initial() > limit {
+					Some("table exceeds maximum size allowed")
+				} else {
+					None
+				}
+			});
+		match err_msg {
+			Some(msg) => Err(msg),
+			None => Ok(())
+		}
```

Comment:
Nit: I'd prefer more clear code to using combinators. Here is an example pseudo-code:

```rust
if let Some(table_type) = ... {
  if table_type.limits().initial() > limit {
    return Err("table exceeds maximum size allowed");
  }
}
Ok(())

Reply:
OK